### PR TITLE
Add proxy capability, fix PASS option, change the README example

### PR DESCRIPTION
**Pull request by gmolveau**
Hi,

I wanted to use your library but a few things were missing so I added some features.

* added a Proxy option  
* changed the scope of some of the Bot struct variables  
* fixed the PASS option to enter password on IRC servers  
* changed the example in the README with a parser and a Tor IRC server

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ## go-irc
 Library to interact with IRC.
 
+Forked from [gianarb/go-irc](https://github.com/gianarb/go-irc)
```
Comment (gianarb) [inline README.md]: Hello

Can you remove this please? :)

**Comment by gianarb** [comment]
Hello @gmolveau ! I happy to hear that! The code looks great, I just have a small request. After that I am ready to merge!

File: README.md [hunk 2]
```diff
@@ -35,48 +34,60 @@ type Message struct {
...
 ```go
 package main
 
 import (
-    "log"
-    "fmt"
-    "regexp"
     "bufio"
+    "fmt"
+    irc "github.com/gmolveau/go-irc"
```
Comment (gmolveau) [inline README.md]: I'll change here too

**Comment by gianarb** [comment]
thanks

**Pull request merged**